src/ex4/dist/scripts/Main.js

```diff
@@ -0,0 +1,304 @@
+import { ItemManager } from "../clients/ItemManager.js";
+import { Alert } from "./Alert.js";
+const EMPTY_INPUT_MSG = "Task input cannot be empty, please try again";
+
+class main {
+  constructor(htmlElement) {
+    this.tasksManager = new ItemManager();
+    this.emprtyInputAlert = new Alert(htmlElement);
+
+    // Action Elements //
+
+    this.taskInput = htmlElement.querySelector(".task-input");
+    this.addTaskButton = htmlElement.getElementById("task-add-btn");
+    this.filterOption = htmlElement.getElementById("filter-dropdown");
+    this.searchInput = htmlElement.getElementById("search-input");
+    this.chosenFilter = "all";
+
+    // Task List Elements //
+
+    this.tasksList = htmlElement.querySelector(".tasks-list");
+    this.dragArea = htmlElement.querySelector(".tasks-list");
+    this.emptyListMsg = htmlElement.getElementById("empty-tasks");
+    this.removeAllTasksButton = htmlElement.getElementById("remove-all-btn");
+
+    // Events init //
+
+    this.activateSortable(this.tasksManager);
+    this.addFieldsEventListeners();
+  }
+
+  // Init Methods //
+
+  activateSortable(tasksManeger) {
+    new Sortable(this.dragArea, {
+      animation: 350,
+      draggable: ".task",
+      handle: ".grip-lines-icon",
+
+      onEnd: function (evt) {
+        const tasksList = evt.item.parentElement.querySelectorAll(".task");
+        tasksManeger.reSortTasks(tasksList);
+      },
+    });
+  }
+
+  addFieldsEventListeners() {
+    this.filterOption.onchange = this.onFilterOptionChange.bind(this);
+    this.searchInput.onkeyup = this.onSearchInputKeyUp.bind(this);
+    this.removeAllTasksButton.onclick = this.onRemoveAllTasksClick.bind(this);
+
+    this.taskInput.onkeyup = (e) => {
+      if (e.key === "Enter") {
+        console.log(e);
+        e.preventDefault();
+        this.onAddTaskClick(e);
+      }
+    };
+
+    this.addTaskButton.addEventListener("click", async (e) => {
+      e.preventDefault();
+      await this.onAddTaskClick(e);
+      return false;
+    });
+  }
+
+  async init() {
+    console.log("Main init");
+    await this.renderTasks();
```

Comment:
it's usually preferred by most js linters that all functions will be defined before they are called. Only commenting this once but there are a few occurrences. I love the order and comments in the file, but you can organize it differently to make it fit convention (and consider splitting it later on).